src/spectral/rulesets/ibm-oas/error-response-schema.js

```diff
@@ -0,0 +1,140 @@
+      errors.push({
+        message: '`errors` field should be an array of error models',
+        path: [...pathToProperties, 'errors']
+      });
+    } else {
+      // `errors` is error model container, so validate the items
+      // are valid error models
+      errors.push(
+        ...validateErrorModelSchema(errorResponseProperties.errors.items, [
+          ...pathToProperties,
+          'errors',
+          'items'
+        ])
+      );
+    }
+  } else if (errorResponseProperties.error) {
+    // expect a single error model, validate error model schema
+    errors.push(
+      ...validateErrorModelSchema(errorResponseProperties.error, [
+        ...pathToProperties,
+        'error'
+      ])
+    );
+  } else {
+    // no `errors` or `error` field, so expect error model at top level of schema
+    errors.push(...validateErrorModelSchema(errorResponseSchema, pathToSchema));
+  }
+  return errors;
+}
+
+function validateErrorModelSchema(errorModelSchema, pathToSchema) {
+  const errors = [];
+  if (!schemaIsObjectWithProperties(errorModelSchema)) {
+    errors.push({
+      message: 'Error Model should be an object with properties',
+      path: pathToSchema
+    });
+  } else {
+    // error model has properties, validate the properties
+    if (!hasCodeField(errorModelSchema.properties)) {
+      errors.push({
+        message:
+          'Error Model should contain `code` field, a snake-case, string error code',
+        path: [...pathToSchema, 'properties', 'code']
+      });
+    }
+    if (!hasMessageField(errorModelSchema.properties)) {
+      errors.push({
+        message: 'Error Model should contain a string, `message`, field',
+        path: [...pathToSchema, 'properties', 'message']
+      });
+    }
+    if (!hasMoreInfoField(errorModelSchema.properties)) {
+      errors.push({
+        message:
+          'Error Model should contain `more_info` field that contains a URL with more info about the error',
+        path: [...pathToSchema, 'properties', 'more_info']
+      });
+    }
+  }
+  return errors;
+}
+
+function validStatusCodeField(errorResponseProperties) {
+  // valid if no status_code provided or if the provided status_code is an integer
+  return (
+    !errorResponseProperties.status_code ||
+    errorResponseProperties.status_code.type === 'integer'
+  );
+}
+
+function hasCodeField(errorModelSchemaProperties) {
+  return (
+    errorModelSchemaProperties.code &&
+    errorModelSchemaProperties.code.type === 'string'
+  );
+}
+
+function hasMessageField(errorModelSchemaProperties) {
+  return (
+    errorModelSchemaProperties.message &&
+    errorModelSchemaProperties.message.type === 'string'
+  );
+}
+
+function hasMoreInfoField(errorModelSchemaProperties) {
+  return (
+    errorModelSchemaProperties.more_info &&
+    errorModelSchemaProperties.more_info.type === 'string'
+  );
+}
+
+function hasTraceField(errorResponseProperties) {
+  return (
+    errorResponseProperties &&
```

Comment:
(Tiny comment) You don't check for the "definedness" of this in the other `hasWhatever` functions and I don't think it's necessary here